### PR TITLE
`AnnDataSchema` for anndata validation

### PR DESCRIPTION
**Pull request by ordabayevy**

### CODE REVIEW

**Comment by ordabayevy** [comment]
@mbabadi can you have a look if this validation method looks right?

**Comment by mbabadi** [comment]
@ordabayevy looking at it right now.

File: scvid/data/schema.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    Args:
+        adata (AnnData): Reference AnnData object.
+    """
+
+    attrs = ["obs", "obsm", "var", "varm", "var_names", "layers"]
+
+    def __init__(self, adata: AnnData) -> None:
+        self.attr_values = {}
+        for attr in self.attrs:
+            self.attr_values[attr] = getattr(adata, attr)
```
Comment (mbabadi) [inline scvid/data/schema.py:31]: Can you check that `del adata` does not invalidate references to `self.attr_values`?

File: scvid/data/schema.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        >>> adata.obs["batch"] = [2, 3, 4, 5]
+        >>> adata.var["mu"] = ["a", "b", "c"]
+        >>> schema = AnnDataSchema(ref_adata)
+        >>> schema.validate_adata(adata)
+
+    Args:
+        adata (AnnData): Reference AnnData object.
+    """
+
+    attrs = ["obs", "obsm", "var", "varm", "var_names", "layers"]
```
Comment (mbabadi) [inline scvid/data/schema.py:26]: Please add `varp` attr too. It is a dictionary of square `n_var x n_var` matrices. It should identically match.

File: test/test_data.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        dtype=X.dtype,
+        obs=obs,
+        var=var,
+        layers={"L": L},
+        obsm={"M": M},
+        varm={"V": V},
+    )
+
+
+def test_validate_adata(ref_adata, adata):
```
Comment (mbabadi) [inline test/test_data.py:55]: What do you think about adding a couple of failing fixtures and test them for failure, for example:
- `obs` columns permuted
- `obs` column names are different
- `obs` columns names match but dtype of a column is different
- `obs` columns names match but the categories of a `pandas.Categorical` column are different
- `var` column names are in different order

Testing for failure may seem a bit redundant if the `equals` calls behave as expected, but we don't know that. WDYT?

**Review by mbabadi — CHANGES_REQUESTED** [review]

File: scvid/data/schema.py (hunk 1 above)
Comment (ordabayevy) [reply scvid/data/schema.py:31]: Done.

**Review by ordabayevy — COMMENTED** [review]
@mbabadi Added additional tests that you recommended.

File: test/test_data.py (hunk 3 above)
Comment (ordabayevy) [reply test/test_data.py]: I agree, it makes sense to add tests for failure.

**Review by mbabadi — APPROVED** [review]

**Comment by ordabayevy** [comment]
Thanks for reviewing @mbabadi !

**Pull request merged**